.k8s/components/jobs/restore/db.ts

```diff
@@ -1,46 +1,71 @@
+import { ok } from "assert";
+import fs from "fs";
+import { EnvVar } from "kubernetes-models/v1/EnvVar";
+import { Job } from "kubernetes-models/batch/v1/Job";
+import path from "path";
+import { GITLAB_LIKE_ENVIRONMENT_SLUG } from "../../../utils/GITLAB_LIKE_ENVIRONMENT_SLUG";
+import { PG_ENVIRONMENT_SLUG } from "../../../utils/PG_ENVIRONMENT_SLUG";
+import { getDevDatabaseParameters } from "@socialgouv/kosko-charts/components/azure-pg/params";
+import { getDefaultPgParams } from "@socialgouv/kosko-charts/components/azure-pg";
 
-ok(process.env.BACKUP_DB_NAME);
-ok(process.env.BACKUP_DB_OWNER);
-ok(process.env.BACKUP_DB_FILE);
+const suffix = PG_ENVIRONMENT_SLUG;
+const pgParams = getDevDatabaseParameters({ suffix });
 
 const manifests = restoreDbJob({
-  project: "cdtn-admin",
   env: [
     new EnvVar({
       name: "PGDATABASE",
-      value: process.env.BACKUP_DB_NAME,
+      value: pgParams.database,
     }),
     new EnvVar({
       name: "OWNER",
-      value: process.env.BACKUP_DB_OWNER,
+      value: pgParams.user,
     }),
     new EnvVar({
       name: "FILE",
-      value: process.env.BACKUP_DB_FILE,
+      value: "hasura_prod_db.psql.gz",
     }),
   ],
   postRestoreScript: fs
     .readFileSync(path.join(__dirname, "./post-restore.sql"))
     .toString(),
+  project: "cdtn-admin",
 });
-
 // override initContainer PGDATABASE/PGPASSWORD because this project pipeline use the legacy `db_SHA` convention instead of `autodevops_SHA`
-const job = manifests.find((m) => m.kind === "Job");
-if (job) {
-  //@ts-expect-error
-  const initContainer = job.spec.template.spec.initContainers[0];
-  initContainer.env.find((e: EnvVar) => e.name === "PGDATABASE").value =
-    process.env.BACKUP_DB_NAME;
-  initContainer.env.find(
-    (e: EnvVar) => e.name === "PGPASSWORD"
-  ).value = `pass_${process.env.CI_COMMIT_SHORT_SHA}`;
-}
+const job = manifests.find<Job>((m): m is Job => m.kind === "Job");
+ok(job?.metadata, "Missing job metadata");
+job.metadata.name = `restore-db-${GITLAB_LIKE_ENVIRONMENT_SLUG}`;
+job.metadata!.annotations = {
+  "kapp.k14s.io/update-strategy": "always-replace",
+};
+job.spec!.template!.metadata!.annotations = {
+  "kapp.k14s.io/deploy-logs": "for-new-or-existing",
+};
+ok(
+  job.spec?.template.spec?.initContainers![0].env,
+  "Missing initContainer definition"
+);
+
+const initContainer = job.spec.template.spec.initContainers[0];
+
+const pgDatabaseEnvVar = initContainer.env?.find(
+  (e) => e.name === "PGDATABASE"
+);
+ok(pgDatabaseEnvVar, "Missing PGDATABASE variable");
+pgDatabaseEnvVar.value = pgParams.database;
+
+const pgUserEnvVar = initContainer.env?.find((e) => e.name === "PGUSER");
+const { host } = getDefaultPgParams();
+ok(pgUserEnvVar, "Missing PGUSER variable");
+pgUserEnvVar.value = `${pgParams.user}@${host}`;
+
+const pgPasswordEnvVar = initContainer.env?.find(
+  (e) => e.name === "PGPASSWORD"
+);
+ok(pgPasswordEnvVar, "Missing PGPASSWORD variable");
+pgPasswordEnvVar.value = pgParams.password;
```

Comment:
the initContainer should already have the correct `envFrom.secretRef` no ?

Reply:
This is part of the "one db per branch" hack as the password here is the `azure-pg-user-<branch_name>` password

Reply:
ok so i guess this should disappear soon :)